patch from Debian with changes to configure script and whitespace changes
removed to make the file smaller; original informations below

Description: Perl 5.14 transition, use given/when instead of switch/case
 switch/case are modified to given/when in pacpl, configure.ac edited
 to remove "Switch" perl module check, and then, configure was regenerated
 by autoconf and included in the patch.
Author: Maxime Chatelle <mmyc@gmx.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=629464
Last-Update: 2011-06-23
Forwarded: yes, https://sourceforge.net/tracker/?func=detail&aid=3325345&group_id=139264&atid=742827
diff -Naurb pacpl-4.0.5-orig/configure.ac pacpl-4.0.5/configure.ac
--- pacpl-4.0.5-orig/configure.ac	2012-07-22 02:32:23.765040550 +0200
+++ pacpl-4.0.5/configure.ac	2012-07-22 02:32:35.437524062 +0200
@@ -61,8 +61,7 @@
 echo "checking for required perl modules..."
 echo ""
 
-AC_PROG_PERL_MODULES(	Switch
-			strict
+AC_PROG_PERL_MODULES(   strict
 			warnings 
 			Getopt::Long 
 			File::Basename 
diff -Naurb pacpl-4.0.5-orig/pacpl pacpl-4.0.5/pacpl
--- pacpl-4.0.5-orig/pacpl	2012-07-22 02:32:23.765040550 +0200
+++ pacpl-4.0.5/pacpl	2012-07-22 02:32:35.418521646 +0200
@@ -19,7 +19,7 @@
 
 use strict;
 use warnings;
-use Switch;
+use feature qw(switch);
 use Getopt::Long;
 use File::Basename;
 use File::Find;
@@ -1913,99 +1913,99 @@
 
     my $opt = shift;
 
-      switch ($opt) {
+      given ($opt) {
 
-            case 'BITRATE'   { 
+	  when ('BITRATE')   {
                                 $bitrate  = `$kdialog --default $bitrate  --combobox \"$lang{kde_bitrate}\" 56 112 128 160 192 256 320`;
                                 chomp($bitrate);
                                 exit(1) if not $bitrate;
                              }
 
-            case 'FREQ'      {
+	  when ('FREQ')      {
                                 $freq     = `$kdialog --default $freq     --combobox \"$lang{kde_freq}\" 32000 44100 48000`;
                                 chomp($freq);
                                 exit(1) if not $freq;
                              }
 
-            case 'CHANNELS'  { 
+	  when ('CHANNELS')  {
                                 $channels = `$kdialog --default $channels --combobox \"$lang{kde_chans}\" 1 2`;
                                 chomp($channels);
                                 exit(1) if not $channels;
                              }
 
-            case 'FCOMP'     {
+	  when ('FCOMP')     {
                                 $fcomp    = `$kdialog --default $fcomp    --combobox \"$lang{kde_fcomp}\" 0 1 2 3 4 5 6 7 8`;
                                 chomp($fcomp);
                                 exit(1) if not $fcomp;
                              }
 
-            case 'PCOMP'     {
+	  when ('PCOMP')     {
                                 $pcomp    = `$kdialog --default $pcomp    --combobox \"$lang{kde_pcomp}\" 1 2 3 4 5`;
                                 chomp($pcomp);
                                 exit(1) if not $pcomp;
                              }
 
-            case 'ACOMP'     {
+	  when ('ACOMP')     {
                                 $acomp    = `$kdialog --default $acomp    --combobox \"$lang{kde_acomp}\" 1000 2000 3000 4000 5000`;
                                 chomp($acomp);
                                 exit(1) if not $acomp;
                              }
 
-            case 'OGGQUAL'   {
+	  when ('OGGQUAL')   {
                                $oggqual  = `$kdialog --default $oggqual   --combobox \"$lang{kde_oggqual}\" 0 1 2 3 4 5 6 7 8 9 10`;
                                chomp($oggqual);
                                exit(1) if not $oggqual;
                              }
-                             
-            case 'SPXQUAL'   {
+	  when ('SPXQUAL')   {
                                $spxqual  = `$kdialog --default $spxqual   --combobox \"$lang{kde_spxqual}\" 0 1 2 3 4 5 6 7 8 9 10`;
                                chomp($spxqual);
                                exit(1) if not $spxqual;
                              }
 
-            case 'AACQUAL'   {
+	  when ('AACQUAL')   {
                                $aacqual  = `$kdialog --default $aacqual   --combobox \"$lang{kde_aacqual}\" 50 75 100 125 150`;
                                chomp($aacqual);
                                exit(1) if not $aacqual;
                              }
 
-            case 'MPCQUAL'   {
+	  when ('MPCQUAL')   {
                                $mpcqual  = `$kdialog --default $mpcqual   --combobox \"$lang{kde_mpcqual}\" thumb radio standard xtreme insane braindead`;
                                chomp($mpcqual);
                                exit(1) if not $mpcqual;
                              }
 
-            case 'OFMODE'    {
+	  when ('OFMODE')    {
                                $ofmode   = `$kdialog --default $ofmode    --combobox \"$lang{kde_ofmode}\" fast normal high extra best highnew extranew bestnew`; 
                                chomp($ofmode);
                                exit(1) if not $ofmode;
                              }
 
-            case 'OFOPT'     {
+	  when ('OFOPT')     {
                                $ofopt    = `$kdialog --default $ofopt     --combobox \"$lang{kde_ofopt}\" none fast normal high best`; 
                                chomp($ofopt);
                                exit(1) if not $ofopt;
                              }
 
-            case 'BRATIO'    {
+	  when ('BRATIO')    {
                                $bratio   = `$kdialog --default $bratio    --combobox \"$lang{kde_bratio}\" 1 2 3 4 5`;
                                chomp($bratio);
                                exit(1) if not $bratio;
                              }
 
-            case 'BQUANL'    {
+	  when ('BQUANL')    {
                                $bquanl   = `$kdialog --default $bquanl    --combobox \"$lang{kde_bquanl}\" 1.00 1.25 1.50 1.75 2.00`;
                                chomp($bquanl);
                                exit(1) if not $bquanl;
                              }
 
-            case 'BPSIZE'    {
+	  when ('BPSIZE')    {
                                $bpsize   = `$kdialog --default $bpsize    --combobox \"$lang{kde_bpsize}\" 10 128`;
                                chomp($bpsize);
                                exit(1) if not $bpsize;
                              }
-
-                        else { return 0; }
+	  default {
+	      return 0;
+	  }
       }
 
 }
@@ -2042,28 +2042,23 @@
 
     my $type = shift;
 
-    switch ($type) {
-        
-        case /mp4|m4a|m4b/ { 
-        
+    given ($type) {
+	when (/mp4|m4a|m4b/) {
             if ($run{$to}{ENCODER}{$encoder}{NAME} eq "ffmpeg") {
-     
                 $tag_name{track} = 0 if not $tag_name{track};
                 $tag_name{year}  = 0 if not $tag_name{year};
                 return "-title \"$tag_name{title}\" -track \"$tag_name{track}\" -author \"$tag_name{artist}\" -album \"$tag_name{album}\" -comment \"$tag_name{comment}\" -year \"$tag_name{year}\" -genre \"$tag_name{genre}\"";
-
             } else {
-                
               return "--title \"$tag_name{title}\" --track \"$tag_name{track}\" --artist \"$tag_name{artist}\" --album \"$tag_name{album}\" --comment \"$tag_name{comment}\" --year \"$tag_name{year}\" --genre \"$tag_name{genre}\"";
-
               }
         } 
-                
-        case /mpc|mpp/ { return "--tag \"Title=$tag_name{title}\" --tag \"Track=$tag_name{track}\" --tag \"Artist=$tag_name{artist}\" --tag \"Album=$tag_name{album}\" --tag \"Comment=$tag_name{comment}\" --tag \"Year=$tag_name{year}\" --tag \"Genre=$tag_name{genre}\""; }
-        case 'wv'      { return "-w \"Title=$tag_name{title}\" -w \"Track=$tag_name{track}\" -w \"Artist=$tag_name{artist}\" -w \"Album=$tag_name{album}\" -w \"Comment=$tag_name{comment}\" -w \"Year=$tag_name{year}\" -w \"Genre=$tag_name{genre}\""; }
-        case 'bonk'    { return "-t \"$tag_name{title}\" -a \"$tag_name{artist}\" -c \"$tag_name{comment}\""; }
-        case 'spx'     { return "--author \"$tag_name{artist}\" --title \"$tag_name{title}\" --comment \"date=$tag_name{year}\" --comment \"track=$tag_name{track}\" --comment \"album=$tag_name{album}\" --comment \"genre=$tag_name{genre}\" --comment \"comment=$tag_name{comment}\""; }
-        else           { return ''; }
+	when (/mpc|mpp/) { return "--tag \"Title=$tag_name{title}\" --tag \"Track=$tag_name{track}\" --tag \"Artist=$tag_name{artist}\" --tag \"Album=$tag_name{album}\" --tag \"Comment=$tag_name{comment}\" --tag \"Year=$tag_name{year}\" --tag \"Genre=$tag_name{genre}\""; }
+        when ('wv')      { return "-w \"Title=$tag_name{title}\" -w \"Track=$tag_name{track}\" -w \"Artist=$tag_name{artist}\" -w \"Album=$tag_name{album}\" -w \"Comment=$tag_name{comment}\" -w \"Year=$tag_name{year}\" -w \"Genre=$tag_name{genre}\""; }
+        when ('bonk')    { return "-t \"$tag_name{title}\" -a \"$tag_name{artist}\" -c \"$tag_name{comment}\""; }
+        when ('spx')     { return "--author \"$tag_name{artist}\" --title \"$tag_name{title}\" --comment \"date=$tag_name{year}\" --comment \"track=$tag_name{track}\" --comment \"album=$tag_name{album}\" --comment \"genre=$tag_name{genre}\" --comment \"comment=$tag_name{comment}\""; }
+	default {
+	    return '';
+	}
     }
 }
 
@@ -2075,9 +2070,9 @@
     my ($in_file, $in_format) = @_;
     my $tag_module = $run{$in_format}{TAGS}{MODULE};
 
-    switch ($in_format) {
+    given ($in_format) {
 
-           case 'mp3'  {
+           when ('mp3')  {
                             my $mp3tag = $tag_module->new($in_file);
                             my @tags   = $mp3tag->autoinfo();
 
@@ -2092,7 +2087,7 @@
                             return 0;
                        }
 
-           case 'ogg'  {
+           when ('ogg')  {
                             my $ogg_tag = $tag_module->new($in_file);
 
                             my ($tit, $tra, $art, $alb, $com, $yea, $gen);
@@ -2119,7 +2114,7 @@
                             return 0;
                        }
                        
-        case 'spx'     {
+        when ('spx')     {
                             my $pid  = open(SPX, "speexdec \"$in_file\" tmp1-$$.wav 2>&1 |") or perror("opening_file","$in_file");
                             my @data = <SPX>;
                                @data = grep(!/^$|^#/, @data);
@@ -2144,7 +2139,7 @@
                              return 0;
                        }
 
-       case /flac|fla/ {
+       when (/flac|fla/) {
                             my $flac_tag = $tag_module->new($in_file)->tags();
 
                                $tag_name{title}   = $flac_tag->{TITLE}       if $flac_tag->{TITLE};
@@ -2158,7 +2153,7 @@
                             return 0;
                        }
 
-    case /mp4|m4a|m4b/ {    print "INFILE: $in_file\n";
+    when (/mp4|m4a|m4b/) {    print "INFILE: $in_file\n";
                             my $mp4_tag = $tag_module->new($in_file);
 
                                $tag_name{title}   = $mp4_tag->{TITLE}    if $mp4_tag->{TITLE};
@@ -2172,7 +2167,7 @@
                             return 0;
                        }
 
-        case /mpc|mpp/ {
+        when (/mpc|mpp/) {
 
                             my $mpc_tag = $tag_module->new($in_file)->tags();
 
@@ -2187,7 +2182,7 @@
                             return 0;
                        }
 
-        case 'wv'      {
+        when ('wv')      {
                             my $wv_tag = $tag_module->getTags($in_file)->{'tags'};
 
                                $tag_name{title}   = $wv_tag->{TITLE}   if $wv_tag->{TITLE};
@@ -2201,7 +2196,7 @@
                             return 0;
                        }
 
-        case 'wma'     {
+        when ('wma')     {
                             my $wma_tag = Audio::WMA->new($in_file)->tags();
 
                                $tag_name{title}   = $wma_tag->{TITLE}       if $wma_tag->{TITLE};
@@ -2215,7 +2210,7 @@
                             return 0;
                        }
 
-                  else { return 1; }
+	   default { return 1; }
     }
 }
 
@@ -2225,9 +2220,9 @@
     my ($out_file, $out_format) = @_;
     my  $tag_m;
 
-    switch ($out_format) {
+    given ($out_format) {
 
-           case 'mp3'  {
+           when ('mp3')  {
                             $tag_m = MP3::Tag->new("$out_file");
                             
                             # ID3v2 Tags
@@ -2260,7 +2255,7 @@
 
                         }
 
-           case 'ogg'   {
+           when ('ogg')   {
                             $tag_m = Ogg::Vorbis::Header->new("$out_file");
                             
                             # this prevents duplicate tags =)
@@ -2281,7 +2276,7 @@
 
                         }
 
-       case /fla|flac/ {
+       when (/fla|flac/) {
                             $tag_m = Audio::FLAC::Header->new("$out_file");
                          my $tag_i = $tag_m->tags();
 
@@ -2298,7 +2293,7 @@
                           return 0;
                        }
 
-        else	       { perror("no_write_tag","$out_file"); }
+        default	       { perror("no_write_tag","$out_file"); }
     }
 
 }
